qiskit_nature/second_q/properties/occupied_modals.py

```diff
@@ -41,7 +41,23 @@ def __init__(
                 through which to map the integrals into second quantization. This attribute **MUST**
                 be set before the second-quantized operator can be constructed.
         """
-        super().__init__(self.__class__.__name__, basis)
+        super().__init__(self.__class__.__name__)
+        self._basis: VibrationalBasis = basis
+
+    @property
+    def basis(self) -> VibrationalBasis:
```

Comment:
This will also get refactored further, once the `basis` information can be stored on the `VibrationalStructureProblem` directly (requires `PolyomialTensor`, just like before in the electronic case)